### ISSUE TITLE
Vertical line does not show all the dots

### ISSUE THREAD

**Issue by samwzm** [n3-charts/line-chart#452]
Hi, there,

I have the data sharing the same time, which would be a vertical bar. however, it seems only the bottom data point displays the dot, all the other data points do not display dots. I paste the sample html page below. Could anyone take a look?

By the way, good news is that if I put my mouse over that vertical bar, then all the dots will display.

thanks,
Sam

below is the sample html to show the issue:

<!DOCTYPE html>
<html ng-app="app">
<script src="http://ajax.googleapis.com/ajax/libs/angularjs/1.4.8/angular.min.js"></script>

    <script src="./d3.min.js"></script>
    
    <script src="./LineChart.js"></script>
    <link rel="stylesheet" href="./LineChart.css">

	<script type="text/javascript">
      angular.module('app', ['n3-line-chart'])
    </script>

  <body ng-controller='MainCtrl'>
    <div class="my-chart" hight>
      <linechart data="data" options="options"></linechart>
    </div>

    <script type="text/javascript">

      angular
        .module('app', ['n3-line-chart'])

        .controller('MainCtrl', function($scope) {
          $scope.data = {
            dataset0: [
              {x: 0, val_0: 0, val_1: 0, val_2: 0, val_3: 0},
			  {x: 0, val_0: 0, val_1: 0, val_2: 2, val_3: 0},
              {x: 1, val_0: 0.993, val_1: 3.894, val_2: 6.47, val_3: 14.347},
			  {x: 1, val_0: 0.993, val_1: 3.894, val_2: 8.47, val_3: 14.347},
			  {x: 1, val_0: 0.993, val_1: 3.894, val_2: 10.47, val_3: 14.347},
              {x: 2, val_0: 1.947, val_1: 7.174, val_2: 13.981, val_3: 19.991},
              {x: 3, val_0: 2.823, val_1: 9.32, val_2: 14.608, val_3: 13.509},
              {x: 4, val_0: 3.587, val_1: 9.996, val_2: 10.132, val_3: -1.167},
              {x: 5, val_0: 4.207, val_1: 9.093, val_2: 2.117, val_3: -15.136},
              {x: 6, val_0: 4.66, val_1: 6.755, val_2: -6.638, val_3: -19.923},
              {x: 7, val_0: 4.927, val_1: 3.35, val_2: -13.074, val_3: -12.625}
            ]
          };

          $scope.options = {
            series: [
              {
                axis: "y",
                dataset: "dataset0",
                key: "val_2",
                label: "An area series",
                color: "#1f77b4",
                type: ['line', 'dot'],
                id: 'mySeries0'
              }
            ],
            axes: {x: {key: "x"}}
          };
        });
    </script>
	
	
	<div ng-app="">
 	<p>Name : <input type="text" ng-model="name"></p>
 	<h1>Hello {{name}}</h1>
</div>

  </body>

  
</html>

**Comment by chaosmail** [comment]
Yeah, currently we use the `x` value of the data point as reference - which does not allow multiple `x` values per series. I have to look into this in more detail if theoretically it could be changed. Thanks for reporting this!

**Comment by Freundschaft** [comment]
+1 I'd also volounteer to check if it would be possible to change that

**Comment by lorem--ipsum** [comment]
Okay so this is not gonna happen. Abscissas need to be distinct in order to keep things meaningful in the code, that's one of the few requirements. We can't implement a charting library that deals with every possible data structure. What's the use case for this data you posted @samwzm anyway ? Because to me it makes very little sense to have several different value for the same series on the same abscissa. I'd rather put that into different series.

**Comment by Freundschaft** [comment]
I think what @samwzm wanted to do is build a scatter plot https://upload.wikimedia.org/wikipedia/commons/thumb/0/0f/Oldfaithful3.png/240px-Oldfaithful3.png

**Comment by lorem--ipsum** [comment]
Ha, I hadn't thought of this. This can be useful indeed !
